Make usage of the settings icon more consistent

afercia opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->
Fixes https://github.com/WordPress/gutenberg/issues/62873

## What?
<!-- In a few words, what is the PR actually doing? -->
This PR only aims to make the usage of the `settings` icon more consistent in the editor UI. This icon should only be used to switch a control to an alternative version, see for example the various margin, padding, spacing controls.

Instead, it was used inconsistently in at least 3 other cases:
- The icon button to open the Fonts Library, now replaced with a button with visible text after https://github.com/WordPress/gutenberg/pull/62129
- The 'View Options' icon button in the data views.
- The 'Display settings' icon button in the Query Loop block toolbar.

Screenshot of the icon, for reference:

![Screenshot 2024-07-01 at 13 46 12](https://github.com/WordPress/gutenberg/assets/1682452/8d081a28-4301-4980-b3ba-edcf27664dd8)

Work on this PR surfaced other issues. Most notably, the Query loop block had the 'Display settings' placed in the toolbar, while it makes more sense to have them in the Inspector together with other similar settings.
Other findings will be tracked in new issues.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Icons should uniquely identify specific features or, at least, _similar_ features.
There is already an icon for the concept of 'settings': the `cog` icon is commonly used for this purpose. It makes sense to reuse popular conventions to make the editor UI more intuitive and less disorienting.


## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
- Moves the Query Loop block 'Display settings' to the inspector.
- Simplifies the  'Items per page', 'Offset', and 'Max pages to show' settings by removing the `BaseControl`s that were wrapping the `NumberControl`s. Not sure why these wrapping `BaseControl`s were used in teh first place but they seem unnecessary. Also, the description of the 'Max pages to show' setting was incorrectly set to the wrapping `BaseControl` instead of the actual `NumberControl`.
- Renames 'Max page to show' (singular 'page') to 'Max pages to show' (plural 'pages')
- Changes the dataviews 'View Options` icon to use the `cog` icon.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

### Old testing instructions

- Select a Query Loop block.
- Make sure 'Inherit query from template' in the Inspector is disabled.
- Observe they Query Loop block toolbar doesn't show the 'Display settings' button any longer.
- Observe the 'Items per page', 'Offset', and 'Max pages to show' settings are now in the Inspector.
- Test all the three settings and observe they work as expected.
- Note: the 'Max pages to show' doesn't update the preview in the editor, not even on trunk. I'll create a separate issue for this.
- Observe the 'Max pages to show' description is correctly associated with the number field via `aria-describedby` and ID attributes.
- Go to the Site Editor > Pages
- Observe the 'View Options' icon button now uses the `cog` icon.

### New testing instructions
- Go to the Site Editor > Pages
- Observe the 'View Options' icon button now uses the `cog` icon.
- Select a Query Loop block.
- Make sure 'Inherit query from template' in the Inspector is disabled.
- Click  'Display settings' in the Query Loop block toolbar.
- Observe the UI within the popover doesn't have relevant visual changes.
- Observe the description of the 'Max pages to show' number field is correctly associated to the input with aria-describedby.


### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

## Screenshots or screencast <!-- if applicable -->

The Query loop block settings moved to the Inspector:

![Screenshot 2024-07-01 at 13 50 21](https://github.com/WordPress/gutenberg/assets/1682452/80086093-3167-4773-ae31-43af15a5be27)

The new cog icon in the data views:

![Screenshot 2024-07-01 at 12 43 25](https://github.com/WordPress/gutenberg/assets/1682452/c04aef48-1461-4db3-90e8-52594e8bdc93)

afercia committed:
Move Query block Display settings from block toolbar to inspector.

afercia committed:
Replace dataviews View options icon with the cog icon.

afercia committed:
Remove no longer necessary test.

ntsekouras commented on test/e2e/specs/editor/various/is-typing.spec.js:
```diff
@@ -36,36 +36,4 @@ test.describe( 'isTyping', () => {
 		await expect( blockToolbarPopover ).toBeHidden();
 	} );
-
-	test( 'should not close the dropdown when typing in it', async ( {
```
@ellatrix do you think we should update this test to something like typing in the `link` dialog or something instead of removing it?

ntsekouras replied:
This test should be kept for sure now, that you have reverted the Query Loop changes..

ntsekouras reviewed (commented):
Thanks for the PR! Code wise this looks good. It would be good to have some design input though, especially about the moved controls in inspector controls.  --cc @WordPress/gutenberg-design 

Additionally, if the [change](https://github.com/WordPress/gutenberg/pull/63020/files#diff-35311dc3d34ca7575b2aabdc330138890fcd42f86ee21d23f8405ff5dc9e486cR317)  of icon in DataViews requires more discussion for some reason, we can always split this PR and land the Query Loop changes first.

t-hamano reviewed (commented):
> Moves the Query Loop block 'Display settings' to the inspector.

Refactoring the Query Loop block controls feels like a secondary effort to avoid using icons.

#58207 has already been submitted for refactoring the Query Loop block controls, so can we just discuss the DataViews icon in this PR?

annezazu commented:
>https://github.com/WordPress/gutenberg/pull/58207 has already been submitted for refactoring the Query Loop block controls, so can we just discuss the DataViews icon in this PR?

+1 to this too. I'd rather see that previous PR picked up again to ensure it aligns design wise.

afercia commented:
Thanks for pointing me at https://github.com/WordPress/gutenberg/pull/58207

One of th problems in this project is PRs without an associated issue (which is _required_ accordingly to the contributign guidelines) don't help contributors to follow all the fast-moving parts in this project. 
@youknowriad https://github.com/WordPress/gutenberg/pull/58207 is actually one more PR without assoicated issue, following our conversation at WCEU.

On the specifics od the design proposed in https://github.com/WordPress/gutenberg/pull/58207 I will comment on that PR but I'm not sure I like hte idea of burying down into an ellispis menu such important controls.

afercia commented:
> Additionally, if the [change](https://github.com/WordPress/gutenberg/pull/63020/files#diff-35311dc3d34ca7575b2aabdc330138890fcd42f86ee21d23f8405ff5dc9e486cR317) of icon in DataViews requires more discussion for some reason [...]

I perceived [this comment on the issue](https://github.com/WordPress/gutenberg/issues/62873#issuecomment-2199854377) as a green light to use the `cog` icon.

priethor commented:
> One of th problems in this project is PRs without an associated issue (which is required accordingly to the contributign guidelines) don't help contributors to follow all the fast-moving parts in this project.

I agree for the most part, but there are some exceptional cases where a simple PR following up on another one shouldn't need an extra task; that would actually prevent the project from moving fast and create noise. I would be happy to update the contributing guidelines to reflect this point.

afercia commented:
> > I would be happy to update the contributing guidelines to reflect this point.
> 
> Actually, this is not required, as the guidelines specify that only `Non-trivial pull requests should be preceded by a related issue.`

I know. That's why I stated that creating an issue _is required accordingly to the contributing guidelines of this project_ (for non-trivial changes). A while ago I also researched the history of the project to read the conversation and arguments made when the guidelines were introduced in the documentation and I'd totally agree with those argumentations.

> there are some exceptional cases where a simple PR following up on another one shouldn't need an extra task

Sure, in that case the PR could just have a short notation 'Follow up to' with a link to the relevant issue.

afercia committed:
Revert changes to the Query loop block.

afercia committed:
Fix typo.

afercia commented:
I reverted the changes to the Query loop block and fixed the typo `Max page to show` -> `Max pages to show`.

I'll try to help https://github.com/WordPress/gutenberg/pull/58207 move on.

ntsekouras reviewed (approved):
We need to add back the removed test and this looks good to land. Thanks!

afercia committed:
Add back test.

afercia commented:
Oh yes thanks. I also forgot to remove the unnecessary `BaseControl` wrapping the `NumberControl` that prevent aria-describedby to work as expected.

afercia committed:
Remove unnecessary BaseControl and fix aria-describedby.

afercia commented:
Added back the missing test and removed the redundant `BaseControl` components. The only minor visual change is the slightly smaller bottom margin in teh popover after the description. Screenshot before and after:

![before and after](https://github.com/WordPress/gutenberg/assets/1682452/e15e8da6-0569-4692-8342-67f7e2f190e3)

I will update the testing instructions.
